pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -673,10 +704,8 @@ func (c *Reconciler) createTaskRun(ctx context.Context, rprt *resources.Resolved
 		}}
 
 	if rprt.ResolvedTaskResources.TaskName != "" {
-		tr.Spec.TaskRef = &v1beta1.TaskRef{
-			Name: rprt.ResolvedTaskResources.TaskName,
-			Kind: rprt.ResolvedTaskResources.Kind,
-		}
+		// We pass the entire, original task ref because it may contain additional references like a Bundle url.
+		tr.Spec.TaskRef = rprt.PipelineTask.TaskRef
```

Comment:
this change seems a bit weird and might break :thinking: No guarantee that `rprt.PipelineTask.TaskRef` holds the same as `rprt.ResolvedTaskResources.TaskName` and `rprt.ResolvedTaskResources.Kind`, do we ?

Reply:
This is part of the change is to the resolver to embed the original unedited `PipelineTask` into the resolved Task: https://github.com/tektoncd/pipeline/pull/3142/files/ffc949d0a0694087c67475a43c00d12f1dcc64f7#diff-4e301064fef70f3c0db810267c0959f2R432

The reason being that before we always assumed the task was defined in the cluster and that we could always re-fetch it with just the name and the kind. Going forward, there might also be a bundle url (and who knows, maybe other fields) so it is probably safest to reuse the same task ref as the original.

We could try to map all of the properties instead but it feels safer to reuse the original definition presuming the resolver holds up its contract and doesn't edit that definition before handing it back to us.